config/src/main/java/org/springframework/security/config/annotation/web/configurers/ott/OneTimeTokenLoginConfigurer.java

```diff
@@ -0,0 +1,348 @@
+	}
+
+	/**
+	 * Use this {@link AuthenticationConverter} when converting incoming requests to an
+	 * {@link Authentication}. By default, the {@link OneTimeTokenAuthenticationConverter}
+	 * is used.
+	 * @param authenticationConverter the {@link AuthenticationConverter} to use
+	 */
+	public OneTimeTokenLoginConfigurer<H> authenticationConverter(AuthenticationConverter authenticationConverter) {
+		Assert.notNull(authenticationConverter, "authenticationConverter cannot be null");
+		this.authenticationConverter = authenticationConverter;
+		return this;
+	}
+
+	/**
+	 * Specifies the {@link AuthenticationFailureHandler} to use when authentication
+	 * fails. The default is redirecting to "/login?error" using
+	 * {@link SimpleUrlAuthenticationFailureHandler}
+	 * @param authenticationFailureHandler the {@link AuthenticationFailureHandler} to use
+	 * when authentication fails.
+	 */
+	public OneTimeTokenLoginConfigurer<H> authenticationFailureHandler(
+			AuthenticationFailureHandler authenticationFailureHandler) {
+		Assert.notNull(authenticationFailureHandler, "authenticationFailureHandler cannot be null");
+		this.authenticationFailureHandler = authenticationFailureHandler;
+		return this;
+	}
+
+	/**
+	 * Specifies the {@link AuthenticationSuccessHandler} to be used. The default is
+	 * {@link SavedRequestAwareAuthenticationSuccessHandler} with no additional properties
+	 * set.
+	 * @param authenticationSuccessHandler the {@link AuthenticationSuccessHandler}.
+	 */
+	public OneTimeTokenLoginConfigurer<H> authenticationSuccessHandler(
+			AuthenticationSuccessHandler authenticationSuccessHandler) {
+		Assert.notNull(authenticationSuccessHandler, "authenticationSuccessHandler cannot be null");
+		this.authenticationSuccessHandler = authenticationSuccessHandler;
+		return this;
+	}
+
+	private AuthenticationFailureHandler getAuthenticationFailureHandler() {
+		if (this.authenticationFailureHandler != null) {
+			return this.authenticationFailureHandler;
+		}
+		this.authenticationFailureHandler = new SimpleUrlAuthenticationFailureHandler("/login?error");
+		return this.authenticationFailureHandler;
+	}
+
+	private OneTimeTokenService getOneTimeTokenService(H http) {
+		if (this.oneTimeTokenService != null) {
+			return this.oneTimeTokenService;
+		}
+		OneTimeTokenService bean = getBeanOrNull(http, OneTimeTokenService.class);
+		if (bean != null) {
+			this.oneTimeTokenService = bean;
+		}
+		else {
+			this.logger.debug("Configuring InMemoryOneTimeTokenService for oneTimeTokenLogin()");
+			this.oneTimeTokenService = new InMemoryOneTimeTokenService();
+		}
+		return this.oneTimeTokenService;
+	}
+
+	private OneTimeTokenSender getOneTimeTokenSender(H http) {
+		if (this.oneTimeTokenSender != null) {
+			return this.oneTimeTokenSender;
+		}
+		OneTimeTokenSender bean = getBeanOrNull(http, OneTimeTokenSender.class);
+		if (bean == null) {
+			throw new IllegalStateException("A OneTimeTokenSender is required for oneTimeTokenLogin(). "
```

Comment:
`OneTimeTokenSender` has been semantically replaced by the `GeneratedOneTimeTokenSuccessHandler`. I'll resolve this conversation since it refers to an old version of the PR.